src/rules/no-duplicate-dollar-variables/index.js

```diff
@@ -60,7 +95,19 @@ export default function(value, secondaryOptions) {
         return;
       }
 
-      if (vars[decl.prop]) {
+      const ancestors = [];
+      let parent = decl.parent;
+
+      while (parent !== null && parent !== undefined) {
+        const parentKey = parent.toString();
```

Comment:
This seems to be inefficient - I think this serializes the entire subtree below [parent]. I'm not sure what to use as unique keys to traverse down the scopes though. the non-leaf nodes are all [Container](https://api.postcss.org/Container.html)s I think? I couldn't find a good common way to uniquify them...